taxi/tests/test_views.py

```diff
@@ -0,0 +1,123 @@
+        )
+        self.client.login(
+            username="testuser",
+            password="testpass"
+        )
+        self.manufacturer = Manufacturer.objects.create(
+            name="Toyota",
+            country="Japan"
+        )
+        self.car = Car.objects.create(
+            model="Camry",
+            manufacturer=self.manufacturer
+        )
+
+
+class IndexViewTest(BaseTestSetup):
+    def test_index_view(self):
+        response = self.client.get(reverse("taxi:index"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "taxi/index.html")
+        self.assertIn("num_drivers", response.context)
+        self.assertIn("num_cars", response.context)
+        self.assertIn("num_manufacturers", response.context)
+
+
+class ManufacturerListViewTest(BaseTestSetup):
+    def setUp(self):
+        super().setUp()
+        Manufacturer.objects.create(name="Ford", country="USA")
+
+    def test_manufacturer_list_view(self):
+        response = self.client.get(reverse("taxi:manufacturer-list"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
+        self.assertEqual(len(response.context["manufacturer_list"]), 2)
+
+
+class ManufacturerCreateViewTest(BaseTestSetup):
+    def test_manufacturer_create_view(self):
+        response = self.client.post(
+            reverse("taxi:manufacturer-create"),
+            {"name": "BMW", "country": "Germany"}
+        )
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(Manufacturer.objects.filter(name="BMW").exists())
+
+
+class CarListViewTest(BaseTestSetup):
+    def test_car_list_view(self):
+        response = self.client.get(reverse("taxi:car-list"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "taxi/car_list.html")
+        self.assertEqual(len(response.context["car_list"]), 1)
+
+
+class CarDetailViewTest(BaseTestSetup):
+    def test_car_detail_view(self):
+        response = self.client.get(
+            reverse("taxi:car-detail",
+                    args=[self.car.id])
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "taxi/car_detail.html")
+        self.assertEqual(response.context["car"], self.car)
+
+
+class DriverListViewTest(BaseTestSetup):
+    def setUp(self):
+        super().setUp()
+        Driver.objects.create(
+            username="driver1",
+            license_number="ABC123456"
+        )
+        Driver.objects.create(
+            username="driver2",
+            license_number="XYZ987654"
+        )
+
+    def test_driver_list_view(self):
+        response = self.client.get(reverse("taxi:driver-list"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "taxi/driver_list.html")
+        self.assertEqual(len(response.context["driver_list"]), 3)
```

Comment:
Ensure that the expected number of drivers (3) in the `driver_list` context matches the actual number of drivers created in the `setUp` method. Adjust the expected count if additional drivers are added in the setup.